packages/gatsby/src/query/file-parser.js

```diff
@@ -418,7 +418,11 @@ export default class FileParser {
       return null
     }
 
-    if (!text.includes(`graphql`)) return null
+    if (
+      !text.includes(`graphql`) &&
+      !text.includes(`gatsby-plugin-static-image`)
```

Comment:
Do we have to include it? Cause I rather not change anything in gatsby core

Reply:
Unfortunately yes. It does a string check on the source to see if it includes the string "graphql". Before this I was literally just including `"graphql"` in the file to make it pick it up. The other option is to remove that check entirely, but it's an otherwise good optimisation.